### ISSUE TITLE
Break up staging release build job

### ISSUE THREAD

**Issue by jaxesn** [aws/eks-anywhere#1049]
**What would you like to be added**:

Break apart the staging release build into individual jobs which can (mostly) be run in parallel.

**Why is this needed**:

Currently the staging build, correctly, rebuilds all projects to ensure no oddities with dependencies.  This is done via a single code build job.  It would be nice if we could split this up into a pipeline which could run most of the builds in parallel and then 1 (or more) final jobs that put everything together and push the bundle.

**Comment by abhay-krishna** [comment]
This could be a potential usecase for [Codebuild batch builds](https://docs.aws.amazon.com/codebuild/latest/userguide/batch-build.html). We define the dependency tree in the batch build configuration and maintain the same workflow of queueing some projects until their dependencies have been built.
